ProcessLib/Utils/GlobalVectorUtils.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "MeshLib/Properties.h"
+
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+
+namespace ProcessLib
+{
+/// Copies part of a global vector for the given variable into output vector
+/// while applying a function to each value.
+///
+/// \attention The output_vector is accessed through node id and component,
+/// therefore multiple meshes are not supported.
+template <typename Functor>
+void transformVariableFromGlobalVector(
+    GlobalVector const& input_vector, int const variable_id,
+    NumLib::LocalToGlobalIndexMap const& local_to_global_index_map,
+    MeshLib::PropertyVector<double>& output_vector, Functor mapFunction)
+{
+    std::fill(begin(output_vector), end(output_vector),
+              std::numeric_limits<double>::quiet_NaN());
```

Comment:
In HM the generated output for the pressure-related rhs will be a mix of NaNs and proper values, right?
Maybe for that case the rhs should be interpolated to the second-order nodes in that case?

Reply:
Yes, the nodes not participating in the process have NaN output.
Interpolation is not suitable IMO. I try to explain it in the sketch:
```
Force <-----  * ==== x ==== * ==== x ==== * .... ==== x ==== * -----> opposing Force
```
where `*` is a node with nodal force output, `x` is the node with NaN force, and `====` is the element connection.
The sum of the forces is 0 and can occur only on boundary nodes (injection nodes). Interpolation does not necessarily leads to balanced forces, only if the "secondary" nodes lie in the middle between the "primary" nodes.